### PR TITLE
feat: Update middleware.md to fix middleware.matcher example

### PR DESCRIPTION
**Pull request by hydRAnger**
### What?
As the intention of matcher example here, is to match all 'page' routes, but ignore static resources and things like API.
The current regex seems won't match the **root path '/'**?

### How?
Adjust the regex for matching all 'page' routes.
or do you think the below is clearer?
```javascript
export const config = {
  matcher: [
    /*
     * Match all request paths except for the ones starting with:
     * - api (API routes)
     * - _next/static (static files)
     * - _next/image (image optimization files)
     * - favicon.ico (favicon file)
     */
    '/',
    '/((?!api|_next/static|_next/image|favicon.ico).*)',
  ],
};
```

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: f41282a017021f25af20b0d7ba857cfac2a4950b

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: f41282a017021f25af20b0d7ba857cfac2a4950b

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: docs/02-app/01-building-your-application/01-routing/11-middleware.mdx [hunk 1]
```diff
@@ -76,7 +76,7 @@ export const config = {
      * - _next/image (image optimization files)
      * - favicon.ico (favicon file)
      */
-    '/((?!api|_next/static|_next/image|favicon.ico).*)',
+    '/((?!api|_next/static|_next/image|favicon.ico).*)?',
```
Comment (delbaoliveira) [inline docs/02-app/01-building-your-application/01-routing/11-middleware.mdx:79]: Hey @hydRAnger, thanks for your suggestion! 

I think the addition of `?` could end up confusing people as it makes the preceding group optional. 

Although complex, you could try `/^((?!api|_next\/static|_next\/image|favicon.ico).*|\/)$/` to match the root. Let's see if that works for your use case.

**Review by delbaoliveira — CHANGES_REQUESTED** [review]

**Review by ijjk — CHANGES_REQUESTED** [review]
Hi, the matcher as is should already be matching the root: 
```sh
> require('next/dist/compiled/path-to-regexp').pathToRegexp('/((?!api|_next/static|_next/image|favicon.ico).*)').test('/')
true
```
If this isn't working please provide a reproduction and we can investigate further!

**Pull request closed without merging**